Gems/Multiplayer/Code/Source/Debug/MultiplayerDebugByteReporter.cpp

```diff
@@ -0,0 +1,203 @@
+            m_totalBytesLastSecond = k_millisecondsPerSecond * aznumeric_cast<float>(m_totalBytesThisSecond) / aznumeric_cast<float>(msSinceLastUpdate.count());
+            m_totalBytesThisSecond = 0;
+            m_lastUpdateTime = now;
+        }
+
+        constexpr float bitsPerByte = 8.0f;
+        constexpr int bitsPerKilobit = 1024;
+        return bitsPerByte * m_totalBytesLastSecond / bitsPerKilobit;
+    }
+
+    void MultiplayerDebugByteReporter::Combine(const MultiplayerDebugByteReporter& other)
+    {
+        m_count += other.m_count;
+        m_totalBytes += other.m_totalBytes;
+        m_totalBytesThisSecond += other.m_totalBytesThisSecond;
+        m_minBytes = AZStd::GetMin(m_minBytes, other.m_minBytes);
+        m_maxBytes = AZStd::GetMax(m_maxBytes, other.m_maxBytes);
+    }
+
+    void MultiplayerDebugByteReporter::Reset()
+    {
+        m_count = 0;
+        m_totalBytes = 0;
+        m_totalBytesThisSecond = 0;
+        m_totalBytesLastSecond = 0;
+        m_minBytes = std::numeric_limits<decltype(m_minBytes)>::max();
+        m_maxBytes = 0;
+        m_aggregateBytes = 0;
+    }
+
+    void MultiplayerDebugComponentReporter::ReportField(const char* fieldName, size_t byteSize)
+    {
+        MultiplayerDebugByteReporter::AggregateBytes(byteSize);
+        m_fieldReports[fieldName].ReportBytes(byteSize);
+    }
+
+    void MultiplayerDebugComponentReporter::ReportFragmentEnd()
+    {
+        MultiplayerDebugByteReporter::ReportAggregateBytes();
+        m_componentDirtyBytes.ReportAggregateBytes();
+    }
+
+    AZStd::vector<MultiplayerDebugComponentReporter::Report> MultiplayerDebugComponentReporter::GetFieldReports()
+    {
+        AZStd::vector<Report> copy;
+        for (auto field = m_fieldReports.begin(); field != m_fieldReports.end(); ++field)
+        {
+            copy.emplace_back(field->first, &field->second);
+        }
+
+        auto sortByFrequency = [](const Report& a, const Report& b)
+        {
+            return a.second->GetTotalCount() > b.second->GetTotalCount();
+        };
+
+        AZStd::sort(copy.begin(), copy.end(), sortByFrequency);
+
+        return copy;
+    }
+
+    void MultiplayerDebugComponentReporter::Combine(const MultiplayerDebugComponentReporter& other)
+    {
+        MultiplayerDebugByteReporter::Combine(other);
+
+        for (const auto& fieldIterator : other.m_fieldReports)
+        {
+            m_fieldReports[fieldIterator.first].Combine(fieldIterator.second);
+        }
+
+        m_componentDirtyBytes.Combine(other.m_componentDirtyBytes);
+    }
+
+    void MultiplayerDebugEntityReporter::ReportField(AZ::u32 index, const char* componentName,
+        const char* fieldName, size_t byteSize)
+    {
+        if (m_currentComponentReport == nullptr)
+        {
+            std::stringstream component;
+            component << "[" << std::setw(2) << std::setfill('0') << static_cast<int>(index) << "]" << " " << componentName;
```

Comment:
NP: Might want to consider aznumeric_cast here due to standards

Reply:
Sounds good, will do.